core/templates/hashfuncs.h

```diff
@@ -399,6 +283,32 @@ struct HashableHasher {
 	static _FORCE_INLINE_ uint32_t hash(const T &hashable) { return hashable.hash(); }
 };
 
+// Bitwise equality check. Faster than normal. Note: returns true if both values are NaN.
+static _FORCE_INLINE_ bool byte_equal_real(double p_lhs, double p_rhs) {
+	union {
+		double d;
+		uint64_t i;
+	} u1, u2;
+	u1.d = p_lhs;
+	u2.d = p_rhs;
+	uint64_t bytes_lhs = u1.i;
+	uint64_t bytes_rhs = u2.i;
+	return bytes_lhs == bytes_rhs;
```

Comment:
I'd suggest a more compact implementation, such as this one:
```
return *reinterpret_cast<uint64_t*>(&p_lhs) == *reinterpret_cast<uint64_t*>(&p_rhs);
```

To me, this is more readable.

Reply:
I used a similar method from [q_rsqrt](https://en.wikipedia.org/wiki/Fast_inverse_square_root),
```
i = * ( long * ) &y;
```
but the compiler issued a warning due to undefined behavior. And because of this CI did not pass.

Reply:
Then maybe:
`return memcmp(&p_lhs, &p_rhs, sizeof(double)) == 0`

Reply:
Then there will be a function call that will do even worse than before https://godbolt.org/z/1xzYafM15. 

In any case, the Math class uses a similar float-to-int conversion style.
It also can be done like this to reduce the code:
```c
union {
	double d;
	uint64_t i;
} u1, u2;
u1.d = p_lhs;
u2.d = p_rhs;
return u1.i == u2.i;
```